Sources/UtilityBeltNetworking/HTTPClient.swift

```diff
@@ -208,11 +208,10 @@ public class HTTPClient {
                         return .failure(UBNetworkError.unableToDecode(String(describing: T.self), nil))
                     }
                 case let .success(data):
-                    // TODO: Implement mime type checking for JSON before attempting to decode JSON (IOS-1967)
-//                    // If the mime type for the response isn't JSON, we can't decode it
-//                    guard dataResponse.response?.mimeType == "application/json" else {
-//                        return .failure(UBNetworkError.invalidContentType(dataResponse.response?.mimeType ?? "unknown"))
-//                    }
+                    // If the mime type for the response isn't JSON, we can't decode it
+                    guard dataResponse.response?.mimeType == MimeType.json.rawValue else {
+                        return .failure(UBNetworkError.invalidContentType(dataResponse.response?.mimeType ?? "unknown"))
```

Comment:
Why not re-use your validators for this? It doesn't make this call site 

```suggestion
                    // If the mime type for the response isn't JSON, we can't decode it
                    let validator = Validators.ensureMimeType(.json)
                    
                    do {
                        try validator(dataResponse.response?.mimeType)
                    } catch {
                        return .failure(error)
                    }
```

Reply:
In fact, thinking about its ore, I think you could even use the same loop for validators and everything. Any reason not to implement it the same way?

Reply:
ya I like that